pkg/graveler/graveler.go

```diff
@@ -1742,7 +1742,7 @@ func (g *KVGraveler) retryBranchUpdate(ctx context.Context, repository *Reposito
 		}
 		return nil
 	}, bo)
-	if try == setTries {
+	if try > setTries {
```

Comment:
@eden-ohana  There's still a bug on the if statement above this line (1736)
If `err != nil` and `try >= setTries` we return nil - which is incorrect
We should return the error if err != nil regardless of `try`

Reply:
The branch update will returns nil, and then the retry branch update will return an error because of line 1745. It's returning nil to exit the backoff.retry if you return an error it keeps retrying.

Reply:
`if try == setTries` and `err != nil` we return nil - This is a bug

Reply:
done

Reply:
added changes to make it work